### DISCUSSION TITLE
Adguardhome not working in snapshot

### DISCUSSION THREAD

**Discussion by Mummify1867** [immortalwrt/immortalwrt#1182] [Q&A]
### Describe the bug

Regardless of installing adguard via luci or command line, adguardhome does not start.

### ImmortalWrt version

r29020-4773949c81

### ImmortalWrt release

SNAPSHOT

### ImmortalWrt target/subtarget

mediatek/filogic

### Device

GL.iNet GL-MT6000

### Image kind

Official downloaded image

### Steps to reproduce

1 - Install adguard home (line command or luci);
2 - Set service adguardhome enable and service adguardhome start
3 - Go h[ttp://192.168.1.1:3000/](http://192.168.1.1:3000/)

### Actual behaviour

Browser reports page not found

### Expected behaviour

Adguardhome should open the configuration page.

### Additional info

root@ImmortalWrt:~# service
Usage: service <service> [command]
/etc/init.d/adguardhome            enabled         stopped

### Terms

- [X] I am reporting an issue for ImmortalWrt, not an unsupported fork.

**Comment by spothz** [comment]
ImmortalWrt 23.05.1 newifi d2.  same problem.  After installing and configuring dhcp, I can’t get to 192.168.1.1:3000

**Comment by theLIKImk** [comment]
`/usr/bin/AdguardHome -s install`

After running this, it will create a normal service on its own and start it
But ads can't be filtered properly

**Comment by upsangel** [comment]
ImmortalWrt 23.05.2, on armv64, After installing and configuring dhcp, I can’t get to 192.168.1.1:3000

**Comment by imshonechen** [comment]
我的 `ImmortalWrt 23.05.2 r27625-416c8c5c91 ` ，`opkg install adguardhome` 后无法成功启动，也不报错

**Reply by imshonechen** [reply]
> 我的 `ImmortalWrt 23.05.2 r27625-416c8c5c91 ` ，`opkg install adguardhome` 后无法成功启动，也不报错

我解决了，`opkg install adguardhome` 无法成功启动，于是我使用`AdGuardHome -v -c /etc/adguardhome.yaml -w /var/adguardhome --no-check-update` 能成功运行，然后就能进入ip:3000的网页，做了简单的配置并在网页端升级了内核后，关闭AdGuardHome ，我再尝试用`service adguardhome start` 启动，仍然失败。然后我输入了一个`AdGuardHome -h` ，显示了提示命令，
![截图_20240630122308](https://github.com/immortalwrt/immortalwrt/assets/13300254/bd26ecb7-5123-45b9-af81-c3164cf27abe)
这才发现`adguardhome `和`AdGuardHome`不一样，opkg install安装后生成的配置和服务是小写的，而ADH核心的命名是大驼峰的。

**Reply by ClubMidd** [reply]
怎么修改呢？我的也是不提示，如何修改成大写？或者怎么修复？

**Reply by imshonechen** [reply]
> 怎么修改呢？我的也是不提示，如何修改成大写？或者怎么修复？
ssh连接上wrt或wrt页面的终端里登录。然后输入 `AdGuardHome -h` 看看有没有提示

**Reply by upsangel** [reply]
try 
`opkg files <pkg>`
to locate where is the bin
For example, opkg files adguardhome, then you will find where is the adg bin file, then you will be able to run it.

However, I have experienced that the latest Adguard Home version can't be run properly on the latest Immortalwrt. Need to manually downgrade it to older version, where you can find in the previous packages in the immortal download.

**Reply by doge1837** [reply]
> 怎么修改呢？我的也是不提示，如何修改成大写？或者怎么修复？

![image](https://github.com/user-attachments/assets/cb26b10c-4222-4f7f-962b-a00fb412947c)
修改 /etc/config/adguardhome 这个文件 把enabled 改成1 默认是0 不启动， 改完去启动项那里启动一下就可以，后续会自启
![image](https://github.com/user-attachments/assets/1dfd5a7d-2d20-4714-b2c8-557fedafb30d)

**Comment by ClubMidd** [comment]
有提示，但是没有说修改文件名称的提示呀

imshonechen ***@***.***>于2024年9月19日 周四下午4:33写道：

**Reply by imshonechen** [reply]
那就不改，我也不知道怎么改，直接使用`AdGuardHome`名。比如`AdGuardHome -v -c /etc/adguardhome.yaml -w /var/adguardhome --no-check-update`启动ADG+配置，然后访问网页192.168.x.x:8080。

**Reply by doge1837** [reply]
![image](https://github.com/user-attachments/assets/cb26b10c-4222-4f7f-962b-a00fb412947c)
修改 /etc/config/adguardhome 这个文件 把enabled 改成1 默认是0 不启动， 改完去启动项那里启动一下就可以，后续会自启
![image](https://github.com/user-attachments/assets/1dfd5a7d-2d20-4714-b2c8-557fedafb30d)

**Comment by 1715173329** [comment]
The service is managed by UCI in immortalwrt.
Can't you just check /etc/config/adguardhome?

**Comment by lee89tw14** [comment]
此問題延續至24.10

**Reply by upsangel** [reply]
> 此問題延續至24.10

我大約估計到了這個問題的原因，是啓動 script 的不一致

對比官版的openwrt，同樣是opkg adguardhome，結果完全不一樣

官版的會自動自啓動 service ， http 放了3000 端口， yaml文件配置在 /etc/ ，working dir放在了 /var （即 /tmp 的symlink

immortal版並不會自動啓動，-s install, 啓動了不存在 yaml 配置文件，working dir 不會放在 /var （這樣導致immort 的 adguard 在 硬路由上的rom 反覆寫 log 損毀）

正確用法要跟 @imshonechen  的 AdGuardHome -v -c /etc/adguardhome.yaml -w /var/adguardhome --no-check-update 才是正確的啓動方法。

**Reply by lee89tw14** [reply]
對的，但是遺留問題不解決實在是很頭痛，我一開始還以為是我哪裡沒搞對我還裝回原版wrt，後來才在討論區注意到這個狀況。

**Reply by dairidong** [reply]
其实是要像上面说的，修改 /etc/config/adguardhome 里面的 enabled option 为 1，貌似只有 immortalwrt 这个分支有这个配置（不清楚原因），剩下的就按照 OpenWrt 的官方 Wiki 操作就行
